core/src/main/java/org/apache/calcite/rel/core/JoinRelType.java

```diff
@@ -16,13 +16,61 @@
  */
 package org.apache.calcite.rel.core;
 
+import org.apache.calcite.linq4j.CorrelateJoinType;
+
 import java.util.Locale;
 
 /**
  * Enumeration of join types.
  */
 public enum JoinRelType {
-  INNER, LEFT, RIGHT, FULL;
+  /**
+   * Inner join.
+   */
+  INNER,
+
+  /**
+   * Left-outer join.
+   */
+  LEFT,
+
+  /**
+   * Right-outer join.
+   */
+  RIGHT,
```

Comment:
Do we really need both `LEFT` and `RIGHT` join types?

Reply:
We can remove it if we always rewrote a right join to left, but how can we describe a right join type in the plan ?

Reply:
It should be always possible to express a right join as a left. If that's too complicated to do in this PR we can treat it in the future ;)

Reply:
+1 to treat it as future, thx @zabetak